components/tikv_util/src/sys/disk.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+use fail::fail_point;
+use std::sync::atomic::{AtomicBool, Ordering};
+static DISK_FULL: AtomicBool = AtomicBool::new(false);
+
+pub fn set_disk_full() {
+    DISK_FULL.store(true, Ordering::Release);
+}
+
+pub fn clear_disk_full() {
+    DISK_FULL.store(false, Ordering::Release);
+}
+
+pub fn is_disk_full() -> bool {
+    DISK_FULL.load(Ordering::Acquire)
+}
+
+pub fn disk_full_precheck(_store_id: u64) -> bool {
+    fail_point!("disk_full_peer_1", _store_id == 1, |_| true);
+    fail_point!("disk_full_peer_2", _store_id == 2, |_| true);
+    false
+}
```

Comment:
Looks like `disk_full_precheck` is always use as ` disk_full_precheck || is_disk_full`, how about combining these two into one like:
```
pub fn is_disk_full() -> bool {
    fail_point!("disk_full_peer_1", _store_id == 1, |_| true);
    fail_point!("disk_full_peer_2", _store_id == 2, |_| true);
    DISK_FULL.load(Ordering::Acquire)
}
```

Reply:
Both are ok.

Reply:
Maybe not. Because disk_full_precheck only used to ut, if we merge both, then the pd store heartbeat will clear it.

Reply:
And there will be a more obscure logic, that is, in other places where precheck is not needed, you have to pass a store_id to this merged function.